clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go

```diff
@@ -0,0 +1,233 @@
+		if err != nil {
+			return summaries, err
+		}
+
+		err = waitForPods(config.ClientSet, runtimeObjectNamespace, runtimeObjectSelector.String(), "", int(runtimeObjectReplicas), timeout-time.Since(start))
+		if err != nil {
+			return summaries, fmt.Errorf("waiting for %s pods error: %v", kind, err)
+		}
+		runningRuntimeObjects++
+		if namespace == metav1.NamespaceAll {
+			glog.Infof("%s: %d %ss with all pods running", waitForControlledPodsRunningName, runningRuntimeObjects, kind)
+		} else {
+			glog.Infof("%s: %v: %d %ss with all pods running", waitForControlledPodsRunningName, namespace, runningRuntimeObjects, kind)
+		}
+	}
+
+	return summaries, fmt.Errorf("timeout while waiting for %ss to be running in namespace '%v' with labels '%v' and fields '%v' - only %d found running with all pods", kind, namespace, labelSelector, fieldSelector, runningRuntimeObjects)
+}
+
+func getRuntimeObjectForKind(c clientset.Interface, kind, namespace, labelSelector, fieldSelector string) ([]runtime.Object, error) {
+	listOpts := metav1.ListOptions{
+		LabelSelector: labelSelector,
+		FieldSelector: fieldSelector,
+	}
+	runtimeObjectsList := make([]runtime.Object, 0)
+	switch kind {
+	case "ReplicationController":
+		list, err := c.CoreV1().ReplicationControllers(namespace).List(listOpts)
+		if err != nil {
+			return runtimeObjectsList, err
+		}
+		for i := range list.Items {
+			runtimeObjectsList = append(runtimeObjectsList, &list.Items[i])
+		}
+		return runtimeObjectsList, nil
+	case "ReplicaSet":
+		list, err := c.ExtensionsV1beta1().ReplicaSets(namespace).List(listOpts)
+		if err != nil {
+			return runtimeObjectsList, err
+		}
+		for i := range list.Items {
+			runtimeObjectsList = append(runtimeObjectsList, &list.Items[i])
+		}
+		return runtimeObjectsList, nil
+	case "Deployment":
+		list, err := c.ExtensionsV1beta1().Deployments(namespace).List(listOpts)
+		if err != nil {
+			return runtimeObjectsList, err
+		}
+		for i := range list.Items {
+			runtimeObjectsList = append(runtimeObjectsList, &list.Items[i])
+		}
+		return runtimeObjectsList, nil
+	case "DaemonSet":
+		list, err := c.ExtensionsV1beta1().DaemonSets(namespace).List(listOpts)
+		if err != nil {
+			return runtimeObjectsList, err
+		}
+		for i := range list.Items {
+			runtimeObjectsList = append(runtimeObjectsList, &list.Items[i])
+		}
+		return runtimeObjectsList, nil
+	case "Job":
+		list, err := c.BatchV1().Jobs(namespace).List(listOpts)
+		if err != nil {
+			return runtimeObjectsList, err
+		}
+		for i := range list.Items {
+			runtimeObjectsList = append(runtimeObjectsList, &list.Items[i])
+		}
+		return runtimeObjectsList, nil
+	default:
+		return runtimeObjectsList, fmt.Errorf("unsupported kind when getting runtime object: %v", kind)
+	}
+}
+
+func getNamespaceRuntimeObject(obj runtime.Object) (string, error) {
+	switch typed := obj.(type) {
+	case *v1.ReplicationController:
+		return typed.ObjectMeta.Namespace, nil
```

Comment:
use MetaAccessor - you won't need this ugly switch,

Reply:
Done.